### ISSUE TITLE
File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_macosx.a

### ISSUE THREAD

**Issue by Tom3652** [flutter/flutter#124474]
We can't build a macOS app since upgrading to Xcode 14.3 
I am pretty sure that this an apple bug but is there anything to do Flutter side to counter this ?  

Note : it was working fine before using Xcode 14.3

My macos deployment target is `11.0`. 

The error is : `File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_macosx.a`

## Steps to Reproduce

Try to build on `macos` platform and see this error.

**Expected results:** 

The app should compile normally

**Actual results:** 

The app doesn't compile

<details>

```
[✓] Flutter (Channel stable, 3.7.10, on macOS 13.3 22E252 darwin-x64, locale fr-FR)
    • Flutter version 3.7.10 on channel stable at /Users/foxtom/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4b12645012 (5 days ago), 2023-04-03 17:46:48 -0700
    • Engine revision ec975089ac
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/foxtom/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.77.1)
    • VS Code at /Users/foxtom/Desktop/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.3 22E252 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.49

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by Tom3652** [comment]
I had released my MacOS app before Xcode 14.3 was released, so everything was fine and when i tried to build it since then it doesn't compile.

I can't reproduce the error with the counter app unfortunately.

**Comment by darshankawar** [comment]
@Tom3652 
This is fixed as part of https://github.com/flutter/flutter/issues/124340, so please switch to latest master to see the fix working.

I'll close this as duplicate. If you disagree, write in comments and I'll reopen it.
